crates/astria-composer/tests/blackbox/geth_collector.rs

```diff
@@ -103,22 +80,17 @@ async fn invalid_nonce_causes_resubmission_under_different_nonce() {
     )
     .await;
 
-    // Mount a response of 0 to a nonce query
-    let nonce_refetch_guard = mount_abci_query_mock(
-        &test_composer.sequencer,
-        "accounts/nonce",
-        NonceResponse {
-            height: 0,
-            nonce: 1,
-        },
-    )
-    .await;
-
     let expected_rollup_ids = vec![RollupId::from_unhashed_bytes("test1")];
     // Expect nonce 1 again so that the resubmitted tx is accepted
     let valid_nonce_guard =
         mount_broadcast_tx_sync_mock(&test_composer.sequencer, expected_rollup_ids, vec![1]).await;
 
+    // Mount a response of 1 to a nonce query
+    test_composer
```

Comment:
should get a mock guard from the mount helper and then wait for it to be satisfied